rust/bridge/shared/src/jni/convert.rs

```diff
@@ -625,98 +623,94 @@ impl crate::support::Env for &'_ JNIEnv<'_> {
     }
 }
 
+/// A marker for Rust objects exposed as opaque handles (pointers converted to `jlong`).
+pub trait BridgeHandle {}
+
+impl<'a, T: BridgeHandle + 'static> SimpleArgTypeInfo<'a> for &T {
+    type ArgType = ObjectHandle;
+    fn convert_from(_env: &JNIEnv, foreign: Self::ArgType) -> SignalJniResult<Self> {
+        Ok(unsafe { native_handle_cast(foreign) }?)
+    }
+}
+
+impl<'a, T: BridgeHandle + 'static> SimpleArgTypeInfo<'a> for Option<&T> {
+    type ArgType = ObjectHandle;
+    fn convert_from(env: &JNIEnv, foreign: Self::ArgType) -> SignalJniResult<Self> {
+        if foreign == 0 {
+            Ok(None)
+        } else {
+            <&T>::convert_from(env, foreign).map(Some)
+        }
+    }
+}
+
+impl<'a, T: BridgeHandle + 'static> SimpleArgTypeInfo<'a> for &mut T {
+    type ArgType = ObjectHandle;
+    fn convert_from(_env: &JNIEnv, foreign: Self::ArgType) -> SignalJniResult<Self> {
+        unsafe { native_handle_cast(foreign) }
+    }
+}
```

Comment:
Why do all of the above Ts need to be 'static?

Reply:
Probably that should just be a requirement on BridgeHandle: if we're going to put something into a app language wrapper, we don't know how long it's going to live, and so it can't (safely) have references to anything with a local lifetime.